argocd_application - info block does not work

based on the documentation (link is below) I added the wollowing `info` block:
 https://registry.terraform.io/providers/oboukili/argocd/latest/docs/resources/application#info

```
resource "argocd_application" "apps" {
...
  spec {
  ...
    info {
      name  = "release"
      value = var.release_version
    }
  }
}
```

it sucessfully builds the plan but failed on apply stage with following error:

```
Plan: 1 to add, 0 to change, 0 to destroy.
argocd_application.apps[0]: Creating...
╷
│ Error: Request cancelled
│ 
│   with argocd_application.apps[0],
│   on argocd_applications.tf line 1, in resource "argocd_application" "apps":
│    1: resource "argocd_application" "apps" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵
Releasing state lock. This may take a few moments...

Stack trace from the terraform-provider-argocd_v2.1.0 plugin:

panic: interface conversion: interface {} is map[string]interface {}, not map[string]string

goroutine 115 [running]:
github.com/oboukili/terraform-provider-argocd/argocd.expandApplicationInfo(0xc000118460, 0xc000c15f20, 0x330b968, 0x4)
        github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:366 +0x274
github.com/oboukili/terraform-provider-argocd/argocd.expandApplicationSpec(0xc000c2c580, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:39 +0x1a7
github.com/oboukili/terraform-provider-argocd/argocd.expandApplication(0xc000c2c580, 0xc000f694dc, 0x3, 0x0, 0x0, 0xc000f694fa, 0x6, 0x0, 0x0, 0x0, ...)
        github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:21 +0x106
github.com/oboukili/terraform-provider-argocd/argocd.resourceArgoCDApplicationCreate(0x370d8f8, 0xc000c26720, 0xc000c2c580, 0x30cd2c0, 0xc000980360, 0xc000c2fbb0, 0x12f900a, 0xc0004f9ee0)
        github.com/oboukili/terraform-provider-argocd/argocd/resource_argocd_application.go:47 +0xc5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0001b5880, 0x370d888, 0xc000b8e580, 0xc000c2c580, 0x30cd2c0, 0xc000980360, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:330 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0001b5880, 0x370d888, 0xc000b8e580, 0xc0004f5650, 0xc0004f9ee0, 0x30cd2c0, 0xc000980360, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:456 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000122438, 0x370d888, 0xc000b8e580, 0xc000b90640, 0xc000b8e580, 0x31f1e00, 0xc000b96100)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc000620c60, 0x370d930, 0xc000b8e580, 0xc0004f5420, 0xc000620c60, 0xc000b961e0, 0xc000f7fba0)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x31f1e00, 0xc000620c60, 0x370d930, 0xc000b961e0, 0xc00079fec0, 0x0, 0x370d930, 0xc000b961e0, 0xc000ba2000, 0xf66)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00028a8c0, 0x3726eb8, 0xc000603800, 0xc000b8c6c0, 0xc000340780, 0x4a25780, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.38.0/server.go:1286 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00028a8c0, 0x3726eb8, 0xc000603800, 0xc000b8c6c0, 0x0)
        google.golang.org/grpc@v1.38.0/server.go:1609 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0009e8380, 0xc00028a8c0, 0x3726eb8, 0xc000603800, 0xc000b8c6c0)
        google.golang.org/grpc@v1.38.0/server.go:934 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.38.0/server.go:932 +0x1fd

Error: The terraform-provider-argocd_v2.1.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

ERRO[0055] 1 error occurred:
        * exit status 1
```

--- a/argocd/resource_argocd_application_test.go
+++ b/argocd/resource_argocd_application_test.go
@@ -3,6 +3,7 @@ package argocd
 import (
 	"context"
 	"fmt"
+	"regexp"
 	"strconv"
 	"testing"
 
@@ -496,6 +497,121 @@ func TestAccArgoCDApplication_OptionalPath(t *testing.T) {
 						"argocd_application.directory",
 						"spec.0.source.0.path",
 						".",
+					),
+				),
+			},
+		}})
+}
+
+func TestAccArgoCDApplication_Info(t *testing.T) {
+	name := acctest.RandomWithPrefix("test-acc-info")
+	info := acctest.RandString(8)
+	value := acctest.RandString(8)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccArgoCDApplicationInfo(name, info, value),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(
+						"argocd_application.info",
+						"metadata.0.uid",
+					),
+					resource.TestCheckResourceAttr(
+						"argocd_application.info",
+						"spec.0.info.0.name",
+						info,
+					),
+					resource.TestCheckResourceAttr(
+						"argocd_application.info",
+						"spec.0.info.0.value",
+						value,
+					),
+				),
+			},
+			{
+				Config: testAccArgoCDApplicationInfoNoName(name, value),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(
+						"argocd_application.info",
+						"metadata.0.uid",
+					),
+					resource.TestCheckResourceAttr(
+						"argocd_application.info",
+						"spec.0.info.0.name",
+						"",
+					),
+					resource.TestCheckResourceAttr(
+						"argocd_application.info",
+						"spec.0.info.0.value",
+						value,
+					),
+				),
+			},
+			{
+				Config: testAccArgoCDApplicationInfoNoValue(name, info),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(
+						"argocd_application.info",
+						"metadata.0.uid",
+					),
+					resource.TestCheckResourceAttr(
+						"argocd_application.info",
+						"spec.0.info.0.name",
+						info,
+					),
+					resource.TestCheckResourceAttr(
+						"argocd_application.info",
+						"spec.0.info.0.value",
+						"",
+					),
+				),
+			},
+			{
+				Config: testAccArgoCDApplicationInfo(name, info, value),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(
+						"argocd_application.info",
+						"metadata.0.uid",
+					),
+					resource.TestCheckResourceAttr(
+						"argocd_application.info",
+						"spec.0.info.0.name",
+						info,
+					),
+					resource.TestCheckResourceAttr(
+						"argocd_application.info",
+						"spec.0.info.0.value",
+						value,
+					),
+				),
+			},
+			{
+				Config: testAccArgoCDApplicationNoInfo(name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(
+						"argocd_application.info",
+						"metadata.0.uid",
+					),
+					resource.TestCheckNoResourceAttr(
+						"argocd_application.info",
+						"spec.0.info",
+					),
+				),
+			},
+			{
+				Config:      testAccArgoCDApplicationInfoEmpty(name),
+				ExpectError: regexp.MustCompile("info: cannot be empty."),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(
+						"argocd_application.info",
+						"metadata.0.uid",
+					),
+					resource.TestCheckNoResourceAttr(
+						"argocd_application.info",
+						"spec.0.info",
 					),
 				),
 			},
@@ -1136,6 +1252,153 @@ resource "argocd_application" "simple" {
   }
 }
 	`, name, revision_history_limit)
+}
+
+func testAccArgoCDApplicationInfo(name, info, value string) string {
+	return fmt.Sprintf(`
+resource "argocd_application" "info" {
+  metadata {
+    name      = "%s"
+    namespace = "argocd"
+    labels = {
+      acceptance = "true"
+    }
+  }
+
+  spec {
+    info {
+      name = "%s"
+      value = "%s"
+    }
+    source {
+      repo_url        = "https://github.com/argoproj/argocd-example-apps"
+      path            = "guestbook"
+      target_revision = "HEAD"
+    }
+
+    destination {
+      server    = "https://kubernetes.default.svc"
+      namespace = "default"
+    }
+  }
+}
+    `, name, info, value)
+}
+
+func testAccArgoCDApplicationInfoNoName(name, value string) string {
+	return fmt.Sprintf(`
+resource "argocd_application" "info" {
+  metadata {
+    name      = "%s"
+    namespace = "argocd"
+    labels = {
+      acceptance = "true"
+    }
+  }
+
+  spec {
+    info {
+      value = "%s"
+    }
+    source {
+      repo_url        = "https://github.com/argoproj/argocd-example-apps"
+      path            = "guestbook"
+      target_revision = "HEAD"
+    }
+
+    destination {
+      server    = "https://kubernetes.default.svc"
+      namespace = "default"
+    }
+  }
+}
+    `, name, value)
+}
+
+func testAccArgoCDApplicationInfoNoValue(name, info string) string {
+	return fmt.Sprintf(`
+resource "argocd_application" "info" {
+  metadata {
+    name      = "%s"
+    namespace = "argocd"
+    labels = {
+      acceptance = "true"
+    }
+  }
+
+  spec {
+    info {
+      name = "%s"
+    }
+    source {
+      repo_url        = "https://github.com/argoproj/argocd-example-apps"
+      path            = "guestbook"
+      target_revision = "HEAD"
+    }
+
+    destination {
+      server    = "https://kubernetes.default.svc"
+      namespace = "default"
+    }
+  }
+}
+    `, name, info)
+}
+
+func testAccArgoCDApplicationInfoEmpty(name string) string {
+	return fmt.Sprintf(`
+resource "argocd_application" "info" {
+  metadata {
+    name      = "%s"
+    namespace = "argocd"
+    labels = {
+      acceptance = "true"
+    }
+  }
+
+  spec {
+    info {
+    }
+    source {
+      repo_url        = "https://github.com/argoproj/argocd-example-apps"
+      path            = "guestbook"
+      target_revision = "HEAD"
+    }
+
+    destination {
+      server    = "https://kubernetes.default.svc"
+      namespace = "default"
+    }
+  }
+}
+    `, name)
+}
+
+func testAccArgoCDApplicationNoInfo(name string) string {
+	return fmt.Sprintf(`
+resource "argocd_application" "info" {
+  metadata {
+    name      = "%s"
+    namespace = "argocd"
+    labels = {
+      acceptance = "true"
+    }
+  }
+
+  spec {
+    source {
+      repo_url        = "https://github.com/argoproj/argocd-example-apps"
+      path            = "guestbook"
+      target_revision = "HEAD"
+    }
+
+    destination {
+      server    = "https://kubernetes.default.svc"
+      namespace = "default"
+    }
+  }
+}
+    `, name)
 }
 
 func testAccSkipFeatureIgnoreDiffJQPathExpressions() (bool, error) {

--- a/argocd/structure_application.go
+++ b/argocd/structure_application.go
@@ -36,13 +36,19 @@ func expandApplicationSpec(d *schema.ResourceData) (
 		spec.RevisionHistoryLimit = &pv
 	}
 	if v, ok := s["info"]; ok {
-		spec.Info = expandApplicationInfo(v.(*schema.Set))
+		spec.Info, diags = expandApplicationInfo(v.(*schema.Set))
+		if len(diags) > 0 {
+			return
+		}
 	}
 	if v, ok := s["ignore_difference"]; ok {
 		spec.IgnoreDifferences = expandApplicationIgnoreDifferences(v.([]interface{}))
 	}
 	if v, ok := s["sync_policy"]; ok {
 		spec.SyncPolicy, diags = expandApplicationSyncPolicy(v.([]interface{}))
+		if len(diags) > 0 {
+			return
+		}
 	}
 	if v, ok := s["destination"]; ok {
 		spec.Destination = expandApplicationDestination(v.(*schema.Set).List()[0])
@@ -379,12 +385,30 @@ func expandApplicationIgnoreDifferences(ids []interface{}) (
 }
 
 func expandApplicationInfo(infos *schema.Set) (
-	result []application.Info) {
+	result []application.Info, diags diag.Diagnostics) {
 	for _, i := range infos.List() {
-		result = append(result, application.Info{
-			Name:  i.(map[string]string)["name"],
-			Value: i.(map[string]string)["value"],
-		})
+		item := i.(map[string]interface{})
+		info := application.Info{}
+		fieldSet := false
+
+		if name, ok := item["name"].(string); ok && name != "" {
+			info.Name = name
+			fieldSet = true
+		}
+		if value, ok := item["value"].(string); ok && value != "" {
+			info.Value = value
+			fieldSet = true
+		}
+
+		if !fieldSet {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "spec.info: cannot be empty. Must only contains 'name' or 'value' fields.",
+			})
+			return
+		}
+
+		result = append(result, info)
 	}
 	return
 }
@@ -480,10 +504,16 @@ func flattenApplicationSyncPolicy(sp *application.SyncPolicy) []map[string]inter
 func flattenApplicationInfo(infos []application.Info) (
 	result []map[string]string) {
 	for _, i := range infos {
-		result = append(result, map[string]string{
-			"name":  i.Name,
-			"value": i.Value,
-		})
+		info := map[string]string{}
+
+		if i.Name != "" {
+			info["name"] = i.Name
+		}
+		if i.Value != "" {
+			info["value"] = i.Value
+		}
+
+		result = append(result, info)
 	}
 	return
 }

